src/main/java/util/TickingTimer.java

```diff
@@ -158,4 +158,13 @@ public boolean isStarted() {
         return started;
     }
 
+    /**
+     * Changes the timer's period and restart the timer based on the new period.
+     * @param periodInSec : the amount of time in second before trigger of the timer.
+     */
+    public synchronized void changePeriod(int periodInSec){
```

Comment:
Based on the comment, the method should be named something like `restart(int newPeriodInSec)`?